zebrad/src/commands/start/sync.rs

```diff
@@ -169,8 +208,9 @@ where
                         "added hashes to download set"
                     );
                 }
-                Ok(r) => tracing::error!("unexpected response {:?}", r),
-                Err(e) => tracing::error!("{:?}", e),
+                Ok(_) => unreachable!("network returned wrong response"),
```

Comment:
#698: this isn't a recoverable error, it's a violation of the `zebra-network` contract. So panic.

Reply:
I thought this was an unavoidable condition thanks to the bitcoin protocol legacy

Reply:
No, the `zebra-network` implementation guarantees that you don't get a response that's unrelated to your request, the only problem is that we expose the network as a service from any type of request to any type of response rather than as N services from this request to this response, etc.

If we had nice combinators like in future tower we could do lightweight conversions of a zebra-network service to a specific service (like mapping a `HashSet<BlockHeaderHash>` into a `Request` and unwrapping the `Response`) but I think exposing multiple services from zebra-network would make the API much more complicated (since you have to convey somehow that the services are all fronting the same peer set).

Reply:
Oh okay, good to know. For some reason I thought that `zebra-network` had been unable to abstract away that upstream issue, glad to know that we're safe from this!

Reply:
Yep, this should probably be better documented because @teor2345 was surprised too in a different context (in that instance it was that ZN guarantees that if you request blocks by hash you'll either get those blocks or you'll get an error, there's no way to get a different unrelated block).

Reply:
#710